src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -697,14 +706,60 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
             None
         };
 
-        let volume_mounts = if self.config.coverage {
-            Some(vec![VolumeMount {
+        let mut volume_mounts = vec![];
+
+        if self.config.coverage {
+            volume_mounts.push(VolumeMount {
                 name: "coverage".to_string(),
                 mount_path: "/coverage".to_string(),
                 ..Default::default()
-            }])
-        } else {
-            None
+            })
+        }
+
+        let volumes = match (disk, &self.config.ephemeral_volume_storage_class) {
+            (true, Some(ephemeral_volume_storage_class)) => {
+                volume_mounts.push(VolumeMount {
+                    name: "scratch-volume".to_string(),
+                    mount_path: "/scratch-directory".to_string(),
+                    ..Default::default()
+                });
+                args.push("--scratch-directory=/scratch-directory".into());
+
+                Some(vec![Volume {
+                    name: "scratch-volume".to_string(),
+                    ephemeral: Some(EphemeralVolumeSource {
+                        volume_claim_template: Some(PersistentVolumeClaimTemplate {
+                            spec: PersistentVolumeClaimSpec {
+                                access_modes: Some(vec!["ReadWriteOnce".to_string()]),
+                                storage_class_name: Some(
+                                    ephemeral_volume_storage_class.to_string(),
+                                ),
+                                resources: Some(ResourceRequirements {
+                                    requests: Some(BTreeMap::from([(
+                                        "storage".to_string(),
+                                        Quantity(
+                                            disk_limit
+                                                .unwrap_or(DiskLimit::ONE_GIB)
+                                                .0
+                                                .as_u64()
+                                                .to_string(),
+                                        ),
+                                    )])),
+                                    ..Default::default()
+                                }),
+                                ..Default::default()
+                            },
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                }])
+            }
+            (true, None) => {
+                unreachable!("asserted in the controller")
```

Comment:
This violates what's meant to be a very strict abstraction boundary between the controller and the orchestrator. The orchestrator is not meant to make any assumptions about how it will be used by the controller. We should return an "internal error: service requested disk but no ephemeral volume storage class was configured" instead.